sdk/java-sdk/src/main/java/kalix/javasdk/view/View.java

```diff
@@ -41,6 +47,33 @@ public void _internalSetUpdateContext(Optional<UpdateContext> context) {
     updateContext = context;
   }
 
+
+  /** INTERNAL API */
+  public void _internalSetViewState(S state) {
+    handlingUpdates = true;
+    viewState = Optional.ofNullable(state);
+  }
+
+  /**
+   * Returns the view state (row) for this invocation as currently stored in Kalix.
+   *
+   * <p>Note that modifying the state directly will not update it in storage. To save the state, one
+   * must call {{@code effects().updateState()}}.
+   *
+   * <p>This method can only be called when handling a command. Calling it outside a method (eg: in
+   * the constructor) will raise a IllegalStateException exception.
+   *
+   * @throws IllegalStateException if accessed outside a handler method
+   */
+  @ApiMayChange
+  protected final S viewState() {
+    // user may call this method inside a command handler and get a null because it's legal
+    // to have emptyState set to null.
+    if (handlingUpdates) return viewState.orElse(null);
+    else
+      throw new IllegalStateException("Current state is only available when handling updates.");
+  }
```

Comment:
If someone has ideas for a better name for this method...

I thought about `currentViewRow` but found it too technical. 

I was also thinking that we should rename `currentState` to `entityState` in the Entities. WDYT?